### ISSUE TITLE
WYSIWYG editor bugs and suggestions

### ISSUE THREAD

**Issue by makoivis** [vaadin/eclipse-plugin-legacy#83]
**Reported by Artur Signell on 7 May 2009 15:57 UTC**
This is a meta ticket for improvements that you think should be made to the WYSIWYG editor. Please add own ideas at the end of this list in the description and put a comment with what you have added.

Tasks
- Rename "Visually Editable Vaadin Component" => "Vaadin CustomComponent (composite)"
- Remove label examples from template class. Template class should be usable as a base for a real component.
- Remove package names from "new component" list for components belonging to the core product.

Problems
- Component tree should scroll to the current component when adding a component and not scroll to the top
- Popups for description and value have broken shadows
- Resizing the editor window inside Eclipse will cause textfields to malfunction. Switching to another application and then back fixes the problem.
- Description value is not always stored when the editor loses focus
- Enter does not always work to trigger a value change event in the property editor / textfields
- The asterisk (*) should be added to the file name only after a change has been performed in the visual designer. Currently it is added immediately when the designer is opened.

Small improvements
- Only characters allowed as variable names should be accepted in the name field
- The name field requires a caption
- The caption field requires a caption
- Width/Height fields should have (W/H) or something else to indicate which is which
- A button for setting component size to 100%/100%
- Undo in the visual designer should undo the last operation. Would require tracking of all operations done in the designer.
- Unused imports should be cleaned after the java file is modified. Maybe organize imports could be used.
- Style name should be some kind of select where standard Vaadin styles are shown (only those applicable to the component) and where user's own styles can be stored.
- Shortcut keys for most common tasks.
- Move component using arrow keys
- Show instructions on how to install XULRunner if it cannot be found. Alternatively include XULRunner automatically somehow.
- Document if it is possible to convert an existing class into a visually editable class and, if so, how it is done.
- Handle invalid layouts in a sensible way.

Larger improvements
- A "test mode" where you can test how the view works, e.g. clicking on buttons to see what happens.
- Improve the "new components" list, for instance by using icons.
- Drag new component to layout (requires drag'n'drop).
  - to AbsoluteLayout
  - to other layouts
- Reorder components within a layout
  - Drag inside the component tree
  - Drag inside the visual representation
- Use the correct application theme

Migrated-From: https://dev.vaadin.com/ticket/2911

**Comment by makoivis** [comment]
**Modified by Artur Signell on 7 May 2009 16:00 UTC**

**Comment by makoivis** [comment]
**Comment by Henri Sara on 8 May 2009 06:08 UTC**
Addressed in [7663]:
- ~~Rename "Visually Editable Vaadin Component" => "Vaadin CustomComponent? (composite)"~~
- ~~Remove label examples from template class. Template class should be usable as a base for a real component.~~

**Comment by makoivis** [comment]
**Comment by Henri Sara on 8 May 2009 09:42 UTC**
Addressed in [7667]:
- ~~Unused imports should be cleaned after the java file is modified. Maybe organize imports could be used.~~

In addition, partly addressed the following by a message pointing to XULRunner installation instructions, but the text is neither a link now copyable yet.
- Show instructions on how to install XULRunner if it cannot be found. Alternatively include XULRunner automatically somehow.

Working copy usage when modifying Java code has been corrected to better handle concurrent modifications.

**Comment by makoivis** [comment]
**Comment by Artur Signell on 11 May 2009 05:42 UTC**
- Marked "Remove package names from "new component" list for components belonging to the core product" as it has apparently been done.
- Moved "Component tree should scroll to the current component when adding a component and not scroll to the top" to improvements section.
- Added problem "Source editor does not work exactly as the built in editor"

**Comment by makoivis** [comment]
**Modified by Artur Signell on 15 May 2009 07:27 UTC**

**Comment by makoivis** [comment]
**Comment by Artur Signell on 15 May 2009 07:32 UTC**
Added field visibility and editing of all fields with simple types.

**Comment by makoivis** [comment]
**Comment by dalton on 20 May 2009 15:22 UTC**
The default names of the components added by the WYSIWYG editor could start with lower case, as any other common Java variable names ([Naming Conventions](http://java.sun.com/docs/codeconv/html/CodeConventions.doc8.html#367)). Currently when you add some component, its default name is e.g. '''`ComboBox_1`''' when it should be '''`comboBox_1`''' or even better '''`comboBox1`'''

**Comment by makoivis** [comment]
**Modified by Artur Signell on 28 May 2009 09:00 UTC**

**Comment by makoivis** [comment]
**Comment by Joonas Lehtinen on 31 May 2009 12:59 UTC**
Added link to another ticket

**Comment by makoivis** [comment]
**Modified by Joonas Lehtinen on 31 May 2009 12:59 UTC**

**Comment by makoivis** [comment]
**Comment by Joonas Lehtinen on 31 May 2009 13:04 UTC**
Moved #3041 and #3042 to the list

**Comment by makoivis** [comment]
**Comment by Joonas Lehtinen on 31 May 2009 13:15 UTC**
Added Layout.SpacingHandler, Layout.MarginHandler editing requirements

**Comment by makoivis** [comment]
**Comment by Joonas Lehtinen on 31 May 2009 13:19 UTC**
In my opinion, it is a bad idea to maintain all these bugs/enhancements in just one ticket as we now have "Eclipse wysiwyg plugin" component in trac. I propose that we split this ticket to proper tickets (one per bug/task) that would allow adding testcases, patches, ... to the tickets.

**Comment by makoivis** [comment]
**Comment by Artur Signell on 3 May 2010 13:43 UTC**
Removed previously fixed issues and issues covered by #4624, #4599, #4601, #4603, #4600.

**Comment by makoivis** [comment]
**Comment by Henri Sara on 21 May 2010 09:04 UTC**
Removed the following obsolete or completed points:
- Description value is not always stored when the editor loses focus - see #3043
- Label XHTML mode is not properly saved to model
- The name field requires a caption
- The caption field requires a caption

Corrected some formatting problems in the description.

**Comment by makoivis** [comment]
**Comment by Artur Signell on 28 Jun 2010 08:55 UTC**
Removed points already covered by tickets.

**Comment by makoivis** [comment]
**Comment by Artur Signell on 28 Jun 2010 09:14 UTC**
Everything converted to tickets.

**Comment by makoivis** [comment]
**Modified by Artur Signell on 23 May 2011 09:18 UTC**

**Comment by makoivis** [comment]
**Modified by Artur Signell on 10 Jun 2011 06:35 UTC**

**Comment by makoivis** [comment]
**Modified by Artur Signell on 12 May 2014 08:40 UTC**

**Comment by makoivis** [comment]
**Comment by Artur Signell on 28 Aug 2014 08:46 UTC**
Milestone Plug-in for Eclipse backlog deleted
